ape_alchemy/provider.py

```diff
@@ -26,6 +28,11 @@ class Alchemy(Web3Provider, UpstreamProvider):
 
     network_uris: Dict[tuple, str] = {}
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.block_page_size = 5000
```

Comment:
Instead of setting these in an `init()`, set them as field values, like this:

```python
class Alchemy(Web3Provider, UpstreamProvider):
    ...

    block_page_size: int = 5000
    concurrency: int = 1
```

Maybe also include comments explaining why we need to override their defaults.
Then, no `init` overload is needed.

Reply:
I was actually thinking these might be best driven via Config instead

Reply:
> I was actually thinking these might be best driven via Config instead

check out this PR on ape https://github.com/ApeWorX/ape/pull/1157#discussion_r1033824686
that was my attempt to make these settings Config-based
i didn't know `ape --config` existed though, so i need to look into it more
judging by the existence of [ConfigWrapper](https://github.com/ApeWorX/ape/blob/45d47f9cc69a4065fc4fe15dd40c9c8cd836592e/src/ape/pytest/config.py#L9), it feels like both the command line and yaml feed into the same destination. adding ape-specific settings is needed in either case, which I think is @unparalleled-js's suggestion below. will look at that next.

Reply:
Yeah, I think @unparalleled-js 's suggestion is ultimately the way to go. we allow plugins to define their own config and make it quite easy to use within the implementation, I think this is exactly the right use case for add'l config